src/hooks/useSubStep.js

```diff
@@ -0,0 +1,58 @@
+import {useState, useRef, useCallback} from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = PropTypes.shape({
+    bodyContent: PropTypes.arrayOf(PropTypes.element).isRequired,
+    startFrom: PropTypes.number.isRequired,
+    onFinished: PropTypes.func.isRequired,
+});
+
+/**
+ * @typedef {Object} ReturnType
+ * @property {React.ReactElement} componentToRender
+ * @property {boolean} isEditing
+ * @property {Function} nextScreen
+ * @property {Function} moveTo
+ */
```

Comment:
Where does this comment belong?

Reply:
not sure, if I understand the question, this typedef defines the return type for the hook so we get nice intellisense when using it. Would you like it to be in some other place?

Reply:
Oh I see now, it's connected to the other JSDOC comment, nice!